commands/operator-sdk/cmd/up/local.go

```diff
@@ -0,0 +1,77 @@
+package up
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/cmdutil"
+	cmdError "github.com/operator-framework/operator-sdk/commands/operator-sdk/error"
+
+	"github.com/spf13/cobra"
+)
+
+func NewLocalCmd() *cobra.Command {
+	upCmd := &cobra.Command{
+		Use:   "local",
+		Short: "Launches the operator locally",
+		Long: `The operator-sdk up local command launches the operator on the local machine 
+by building the operator binary with the ability to access a 
+kubernetes cluster using a kubeconfig file.
+`,
+		Run: upFunc,
+	}
+
+	upCmd.Flags().StringVar(&kubeConfig, "kubeconfig", "", "The file path to kubernetes configuration file; defaults to $HOME/.kube/config")
+
+	return upCmd
+}
+
+var (
+	kubeConfig string
+)
+
+const (
+	gocmd             = "go"
+	run               = "run"
+	cmd               = "cmd"
+	main              = "main.go"
+	defaultConfigPath = ".kube/config"
+)
+
+func upFunc(cmd *cobra.Command, args []string) {
+	mustKubeConfig()
+	cmdutil.MustInProjectRoot()
+	c := cmdutil.GetConfig()
+	up(c.ProjectName)
+}
+
+// mustKubeConfig checks if the kubeconfig file exists.
+func mustKubeConfig() {
+	// if kubeConfig is not specified, search for the default kubeconfig file under the $HOME/.kube/config.
+	if len(kubeConfig) == 0 {
+		usr, err := user.Current()
+		if err != nil {
+			cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to determine user's home dir: %v", err))
+		}
+		kubeConfig = filepath.Join(usr.HomeDir, defaultConfigPath)
+	}
+
+	_, err := os.Stat(kubeConfig)
+	if err != nil && os.IsNotExist(err) {
+		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to fine the Kubernetes config file (%v): %v", kubeConfig, err))
+	}
+}
+
+func up(projectName string) {
+	dc := exec.Command(gocmd, run, filepath.Join(cmd, projectName, main))
+	dc.Stdout = os.Stdout
+	dc.Stderr = os.Stderr
+	dc.Env = append(os.Environ(), fmt.Sprintf("KUBERNETES_CONFIG=%v", kubeConfig))
```

Comment:
Make the string `KUBERNETES_CONFIG` a constant because it's common to pkg/k8sclient/client.go and here.